src/storages/storage_manager.js

```diff
@@ -25,42 +30,40 @@ const MAX_OPENED_STORAGES = 1000;
  * @property {LruCache} cache
  * @ignore
  */
-class StorageManager {
+export default class StorageManager {
     /**
      * @param {T} StorageConstructor
+     * @param {Configuration} [config]
      */
-    constructor(StorageConstructor) {
+    constructor(StorageConstructor, config = Configuration.getDefaults()) {
         this.StorageConstructor = StorageConstructor;
         this.name = StorageConstructor.name;
-        this.cache = cacheContainer.openCache(this.name, MAX_OPENED_STORAGES);
+        this.cache = cacheContainer.openCache(this.name, config.get('maxOpenedStorages'));
+        this.config = config;
     }
 
     /**
-     * @param {string} idOrName
+     * @param {string} [idOrName]
      * @param {object} [options]
      * @param {boolean} [options.forceCloud]
      * @return {Promise<T>}
      */
     async openStorage(idOrName, options = {}) {
-        if (!process.env[ENV_VARS.LOCAL_STORAGE_DIR] && !process.env[ENV_VARS.TOKEN]) {
-            throw new Error(`Cannot use ${this.name} as neither ${ENV_VARS.LOCAL_STORAGE_DIR} nor ${ENV_VARS.TOKEN}`
-                + ' environment variable is set. You need to set one these variables in order to enable data storage.');
-        }
-
-        const isLocal = !!(process.env[ENV_VARS.LOCAL_STORAGE_DIR] && !options.forceCloud);
+        const isLocal = !!(this.config.get('localStorageDir') && !options.forceCloud);
 
         if (!idOrName) {
             const defaultIdEnvVarName = DEFAULT_ID_ENV_VAR_NAMES[this.name];
-            idOrName = process.env[defaultIdEnvVarName];
+            const defaultIdConfigKey = DEFAULT_ID_CONFIG_KEYS[this.name];
+            idOrName = this.config.get(defaultIdConfigKey);
             if (!idOrName && isLocal) idOrName = LOCAL_ENV_VARS[defaultIdEnvVarName];
```

Comment:
The fact that the LOCAL_ENV_VARS are still here is not ideal, what do we need to get rid of those too?

Reply:
I think the local env vars are significant to the `apify-cli` which sets them with `apify run`, but to be honest, I'm not even sure. Most of the stuff are just "default values" that we can easily have in the SDK Config itself.

Reply:
So `LOCAL_ENV_VARS` is now inlined to the `Configuration.DEFAULTS`, and most of the usage of `LOCAL_ENV_VARS` is gone from the code in favour of the config defaults. Only place I kept `LOCAL_ENV_VARS` was the deprecated live server that don't use the config object at all.